src/tools/illink/Directory.Build.targets

```diff
@@ -0,0 +1,96 @@
+<Project>
+  <Import Project="Sdk.targets" Sdk="Microsoft.DotNet.Arcade.Sdk" />
+
+  <Import Project="$(_ToolsProjectTargets)" Condition="Exists('$(_ToolsProjectTargets)')" />
+
+  <Target Name="VSTestIfTestProject">
+    <CallTarget Targets="VSTest" Condition="'$(IsTestProject)' == 'true'" />
+  </Target>
+
+
+  <!-- Shared logic for reference assemblies. This enables apicompat and
+       packaging logic for projects that have corresponding reference assemblies. -->
+  <PropertyGroup>
+    <ContractProject Condition="'$(ContractProject)' == ''">$(MSBuildProjectDirectory)\ref\$(MSBuildProjectName).csproj</ContractProject>
+    <HasMatchingContract Condition="Exists('$(ContractProject)')">true</HasMatchingContract>
+    <RunApiCompat Condition="'$(HasMatchingContract)' != 'true'">false</RunApiCompat>
+    <TargetsForTfmSpecificContentInPackage Condition="'$(HasMatchingContract)' == 'true'">$(TargetsForTfmSpecificContentInPackage);_AddReferenceAssemblyToPackage</TargetsForTfmSpecificContentInPackage>
+  </PropertyGroup>
+  <!-- Work around ApiCompat breaking change https://github.com/dotnet/arcade/issues/5361 -->
+  <ItemGroup Condition="'$(RunApiCompat)' == 'true'">
+    <_DependencyDirectories Include="$(IntermediateOutputPath)" />
+  </ItemGroup>
+  <!-- Add a ProjectReference to the reference assembly project for ApiCompat. -->
+  <!-- This also ensures that it is built when the current project gets packaged. -->
+  <ItemGroup Condition="'$(HasMatchingContract)' == 'true'">
+    <ProjectReference Include="$(ContractProject)">
+      <ReferenceOutputAssembly>false</ReferenceOutputAssembly>
+      <OutputItemType>ResolvedMatchingContract</OutputItemType>
+    </ProjectReference>
+  </ItemGroup>
```

Comment:
All of that ApiCompat infrastructure doesn't work anymore with the new roslyn based ApiCompat as property and items names have changed. Also you aren't referencing ApiCompat anywhere anymore.

As all of this already exists in resolveContract.props and resolveContract.targets, I would just import those two files instead and remove all this code.

We just recently did the same for the CoreLib projects which don't reside under src/libraries either: https://github.com/dotnet/runtime/blob/db213657acc53fc48212867d5728e83d9e36a558/src/coreclr/System.Private.CoreLib/System.Private.CoreLib.csproj#L349-L354